src/psygnal/_group_descriptor.py

```diff
@@ -265,7 +265,12 @@ def _setattr_and_emit_(self: object, name: str, value: Any) -> None:
             group: SignalGroup | None = getattr(self, signal_group_name, None)
             signal: SignalInstance | None = getattr(group, name, None)
             # don't emit if the signal doesn't exist or has no listeners
-            if group is None or signal is None or len(signal) < 2 and not len(group):
+            if (
+                group is None
+                or signal is None
```

Comment:
`isinstance(signal, SignalInstance)`

Reply:
used group.__getitem__ instead, and fixed the __getattr__ issue above that should also fix this